packages/inventory/src/loaders/connection.ts

```diff
@@ -1,9 +1,49 @@
 import {
-  ConfigurableModuleDeclaration,
+  InternalModuleDeclaration,
   LoaderOptions,
+  MODULE_RESOURCE_TYPE,
+  MODULE_SCOPE,
 } from "@medusajs/modules-sdk"
+import { DataSource, DataSourceOptions } from "typeorm"
+
+import * as InventoryModels from "../models"
+import { MedusaError } from "medusa-core-utils"
+import { asValue } from "awilix"
 
 export default async (
-  { configModule }: LoaderOptions,
-  moduleDeclaration?: ConfigurableModuleDeclaration
-): Promise<void> => {}
+  { options, container }: LoaderOptions,
+  moduleDeclaration?: InternalModuleDeclaration
+): Promise<void> => {
+  if (
+    moduleDeclaration?.scope === MODULE_SCOPE.INTERNAL &&
+    moduleDeclaration.resources === MODULE_RESOURCE_TYPE.SHARED
+  ) {
+    return
+  }
+
+  const dbData = options?.database as Record<string, string>
+
+  if (!dbData) {
+    throw new MedusaError(
+      MedusaError.Types.INVALID_ARGUMENT,
+      `Database config is not present at module config "options.database"`
+    )
+  }
+
+  const entities = Object.values(InventoryModels)
+  const dataSource = new DataSource({
```

Comment:
Thougts: One thing to think about is that if the module is isolated and run standalone, the dependencies in package.json will not suffice. So if a modules can be both internal and standalone we have to find a way to manage those deps knowing that we can't just put them all in dependencies to avoid conflicts

Reply:
Awilix and Typeorm should actually be dependencies, not dev dependencies. I guess these are ther only ones needed. Which other dependency you see?

Reply:
I think that’s it as of now ☺️